app/services/markdown_processor/parser.rb

```diff
@@ -127,6 +128,22 @@ def escape_liquid_tags_in_codeblock(content)
       end
     end
 
+    def add_proper_code_tags(content)
+      # return content if there is not a <code> tag or there are triple back ticks
+      return content if /<code>/.match?(content) == false || /```/.match?(content)
+
+      # return content if there is a <pre> and <code> tag
+      return content if /<code>/.match?(content) && /<pre>/.match(content)
+
+      # check if <code> tag is on the same line
+      return content if %r{<code>(.*)</code>}.match?(content)
+
+      # At this point, there should be <code> tags with no wrapping <pre> tag
+      code_with_pre_tag = content.clone
+      code_with_pre_tag.gsub!("<code>", "```")
```

Comment:
`gsub` can take a regex as first argument, so you could replace these three lines here with a single one (no need to `dup` if you use the non-bang version of `gsub`):

```ruby
content.gsub(%r{</?code>}, "```")
```

Seems to work as expected:

```ruby
content = <<EOS
<code>
puts "Hello, world"
</code>
EOS

content.gsub(%r{</?code>}, "```")
#=> "```\nputs \"Hello, world\"\n```\n"
```

Reply:
Completed this as well